idunn/api/search.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from fastapi import Body, Depends
+
+from idunn import settings
+from idunn.geocoder.bragi_client import bragi_client
+from idunn.geocoder.nlu_client import nlu_client, NluClientException
+from idunn.utils import result_filter
+from idunn.instant_answer import normalize
+from ..geocoder.models import ExtraParams, QueryParams, IdunnAutocomplete
+
+logger = logging.getLogger(__name__)
+
+nlu_allowed_languages = settings["NLU_ALLOWED_LANGUAGES"].split(",")
+
+
+def no_search_result(query=None, lang=None):
+    if query is not None:
+        logger.info(
+            "search: no result",
+            extra={
+                "request": {
+                    "query": query,
+                    "lang": lang,
+                }
+            },
+        )
+    return IdunnAutocomplete()
+
+
+async def search(
+    query: QueryParams = Depends(QueryParams), extra: ExtraParams = Body(ExtraParams())
+) -> IdunnAutocomplete:
+    """
+    Perform a query which is intended to display a relevant result directly (as
+    opposed to `autocomplete` which gives a list of plausible results).
+
+    Similarly to `instant_answer`, the result will need some quality checks.
+    """
+    query.q = normalize(query.q)
+
+    if query.lang in nlu_allowed_languages:
```

Comment:
```suggestion
    if query.nlu and query.lang in nlu_allowed_languages:
```

Reply:
We also want to keep this option overridden by [AUTOCOMPLETE_NLU_SHADOW_ENABLED](https://github.com/Qwant/idunn/blob/master/idunn/utils/default_settings.yaml#L130), right?

Reply:
Well, this option has never been actually used, and is probably not very useful now that the `nlu=true` is used by the application in most cases.  
So I think it can be ignored for `/search`.